### PR TITLE
fix mypy failure for new aiohttp

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

File: sdk/core/azure-core/azure/core/pipeline/transport/_base.py [hunk 1]
```diff
@@ -513,7 +513,7 @@ def __init__(self, request, internal_response, block_size=None):
         self.request = request
         self.internal_response = internal_response
         self.status_code = None  # type: Optional[int]
-        self.headers = {}  # type: Dict[str, str]
+        self.headers = {}  # type: Any
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_base.py:516]: I'm to generalize to `MutableMapping[KT, VT]` but not `Any`

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py [hunk 2]
```diff
@@ -263,8 +264,13 @@ class AioHttpTransportResponse(AsyncHttpResponse):
     def __init__(self, request: HttpRequest, aiohttp_response: aiohttp.ClientResponse, block_size=None) -> None:
         super(AioHttpTransportResponse, self).__init__(request, aiohttp_response, block_size=block_size)
         # https://aiohttp.readthedocs.io/en/stable/client_reference.html#aiohttp.ClientResponse
+        headers = aiohttp_response.headers
+        if isinstance(headers, CIMultiDictProxy):
```
Comment (lmazuel) [inline sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py:268]: Why????????

**Review by lmazuel — CHANGES_REQUESTED** [review]
Let's discuss

File: sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py (hunk 2 above)
Comment (xiangyan99) [reply sdk/core/azure-core/azure/core/pipeline/transport/_aiohttp.py]: Seems like I thought it in an over-complicated way.

**Review by lmazuel — APPROVED** [review]

**Pull request merged**